src/main/java/com/adobe/cq/testing/junit/rules/TemporaryContentAuthorGroup.java

```diff
@@ -0,0 +1,166 @@
+import java.util.UUID;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * TemporaryContentAuthorGroup creates a content author group with write permission under /content
+ * and deletes it at the end of the test.
+ * Whether the delete is successful or not is not checked.
+ * The create operation is retried until a timeout is reached.
+ * The total wait time in the {@code before} method can be up to 30s.
+ */
+public class TemporaryContentAuthorGroup extends ExternalResource {
+    private static final Logger LOG = LoggerFactory.getLogger(TemporaryContentAuthorGroup.class);
+    private static final long DEFAULT_TIMEOUT = Duration.ofSeconds(30).toMillis();
+    private static final long DEFAULT_RETRY_DELAY = Duration.ofSeconds(1).toMillis();
+    private static final String CONTENT_NODE = "/content";
+    private final Supplier<SlingClient> creatorSupplier;
+    private final String groupName;
+
+    private final ThreadLocal<CQClient> groupClient;
+
+    /**
+     * Instantiate a new TemporaryContentAuthorGroup rule, to be used with the {@code @Rule} annotation.
+     * @param creatorSupplier supplier for the client used to create the temporary content author group
+     */
+    public TemporaryContentAuthorGroup(Supplier<SlingClient> creatorSupplier) {
+        if (creatorSupplier == null) {
+            throw new NullArgumentException("creatorSupplier is null");
+        }
+
+        this.creatorSupplier = creatorSupplier;
+        this.groupName = createUniqueAuthorizableId("testGroup");
+
+        this.groupClient = new ThreadLocal<>();
+    }
+
+    /**
+     * @return the CQClient matching the temporary content author group
+     */
+    public CQClient getClient() {
+        return this.groupClient.get();
+    }
+
+    /**
+     * @return the group name
+     */
+    public String getGroupName() {
+        return this.groupName;
+    }
+
+    /**
+     * @return a <code>SlingClient</code> Supplier matching the temporary content author group
+     */
+    public Supplier<SlingClient> getClientSupplier() {
+        return this::getClient;
+    }
+
+    @Override
+    protected void before() throws Throwable {
+        CQSecurityClient securityClient = creatorSupplier.get().adaptTo(CQSecurityClient.class);
+        CQPermissions permissionsObj = new CQPermissions(securityClient);
+
+
+        class CreateGroupPolling extends Polling {
+            Group group;
+
+            @Override
+            public Boolean call() throws Exception {
+                group = securityClient.createGroup((groupName), 201);
+                // set permissions
+                permissionsObj.changePermissions(group.getId(), CONTENT_NODE, true, true, true, false,
+                        false, false, false, 200);
+                return true;
+            }
+        }
+
+        CreateGroupPolling p = new CreateGroupPolling();
+        try {
+            p.poll(DEFAULT_TIMEOUT, SECONDS.toMillis(DEFAULT_RETRY_DELAY));
```

Comment:
Sorry, I meant making this configurable as a field on the `CreateGroupPolling` class and propagate that from a field on `TemporaryContentAuthorGroup`. Basically being able to control the total wait time from the caller.
e.g. 
` new TemporaryContentAuthorGroup(supplier).withTimeout(Duration.ofSeconds(30))`